docs/ELF_DLOPEN_METADATA.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Dlopen Metadata for ELF Files
+category: Interfaces
+layout: default
+SPDX-License-Identifier: LGPL-2.1-or-later
+---
+
+# Dlopen Metadata for ELF Files
+
+*Intended audience: hackers working on packaging ELF files that use dlopen to load libraries.*
+
+## Motivation
+
+Using dlopen() to load optional dependencies brings several advantages: programs can gracefully downgrade
+when a library is not available, and the shared library is only loaded in the process (and its constructors
+are run) only when the required feature is actually used. But it also has some drawbacks, and the main one
```

Comment:
"required feature" sounds wrong here, these features are *optional* after all, i.e. definitely not *required". Did you mean to write "requested"?